Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from mantid.api import FileProperty, MatrixWorkspaceProperty, MultipleFileProperty, \
+    NumericAxis, PropertyMode, Progress, PythonAlgorithm, WorkspaceGroup, WorkspaceGroupProperty, \
+    FileAction, AlgorithmFactory
+from mantid.kernel import Direction, EnabledWhenProperty, IntBoundedValidator, LogicOperator, \
+    PropertyCriterion, PropertyManagerProperty, StringListValidator
+
+from mantid.simpleapi import *
+
+from scipy.constants import physical_constants
+import numpy as np
+import math
+
+
+class PolDiffILLReduction(PythonAlgorithm):
+
+    _mode = 'Monochromatic'
+    _method_data_structure = None # measurement method determined from the data
+    _user_method = None
+    _instrument = None
+    _experimentProperties = None
+
+    _DEG_2_RAD =  np.pi / 180.0
+
+    @staticmethod
+    def _max_value_per_detector(ws):
+        max_values = np.zeros(mtd[ws].getItem(0).getNumberOfHistograms())
+        for entry in mtd[ws]:
+            for spectrum_no in range(entry.getNumberOfHistograms()):
+                dataY = entry.readY(spectrum_no)
+                if dataY > max_values[spectrum_no]:
+                    max_values = dataY
+        return max_values
+
+    def category(self):
+        return 'ILL\\Diffraction'
+
+    def summary(self):
+        return 'Performs polarized diffraction data reduction at the ILL.'
+
+    def seeAlso(self):
+        return ['PolDIffILLAutoProcess']
+
+    def name(self):
+        return 'PolDiffILLReduction'
+
+    def validateInputs(self):
```

Comment:
The validation looks too strict to me. Some of what you require as mandatory I think can be left optional if one has measured the corresponding sample.